rust/agents/relayer/src/msg/processor.rs

```diff
@@ -207,16 +167,13 @@ pub(crate) struct MessageProcessorMetrics {
 }
 
 impl MessageProcessorMetrics {
-    pub fn new(
-        metrics: &CoreMetrics,
-        origin: &HyperlaneDomain,
-        destination: &HyperlaneDomain,
-    ) -> Self {
+    pub fn new(metrics: &CoreMetrics, origin: &HyperlaneDomain) -> Self {
         Self {
             processor_loop_gauge: metrics.last_known_message_nonce().with_label_values(&[
                 "processor_loop",
                 origin.name(),
-                destination.name(),
+                // TODO: For some reason we fail if I remove this...
```

Comment:
Yeah it'll expect label values to be the same length as the declared label names https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/hyperlane-base/src/metrics/core.rs#L112

I think the intent of this is to just know the latest nonce for messages to every destination domain. I kinda question how useful this is-- but we should probably only set this for domains that this relayer plans to actually relay to (so maybe in the `if let Some(send_channel) = self.send_channels.get(&message.destination) {`). Wdyt?

We may need a getter in `MessageProcessorMetrics` to get the gauge for a specific destination domain when it's needed